api/src/main/java/org/pmiops/workbench/db/jdbc/ReportingQueryServiceImpl.java

```diff
@@ -23,87 +25,21 @@
 public class ReportingQueryServiceImpl implements ReportingQueryService {
 
   private final JdbcTemplate jdbcTemplate;
+  private final Provider<WorkbenchConfig> workbenchConfigProvider;
 
-  public ReportingQueryServiceImpl(JdbcTemplate jdbcTemplate) {
+  public ReportingQueryServiceImpl(
+      JdbcTemplate jdbcTemplate, Provider<WorkbenchConfig> workbenchConfigProvider) {
     this.jdbcTemplate = jdbcTemplate;
+    this.workbenchConfigProvider = workbenchConfigProvider;
   }
 
+  // It doesn't make sense to have more rows in memory than you want to upload, because the upload
+  // streaming limit of 10000 in BigQuery is  much more than the effective max number of workspaces
```

Comment:
Could you add a comment to the `maxRowsPerInsert` definition in `WorkbenchConfig` noting that it doesn't make sense to use a value about 10K because BQ won't let you?

Also: does it make sense to clamp it here with something like `min(bqAbsoluteLimit, maxRowsPerInsert)` ?

Reply:
that does make sense.